### PR TITLE
Go back to checking figures for their manager in destroy.

### LINKED ISSUE #18163: Figure can not be closed if it has associated Agg canvas
### Bug report

** plt.fignum_exists() returns True after figure is closed**

Code snippet below closes figure after creating `FigureCanvasAgg`, and it works with matplotlib-3.2.1 but fails with matplotlib-3.3.0

**Code for reproduction**

```python
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt

figure = plt.figure()
plt.axis('scaled')
plt.tight_layout()
matplotlib.backends.backend_agg.FigureCanvasAgg(figure)
plt.close(figure)
print("Matplotlib version:", matplotlib.__version__)
assert plt.fignum_exists(figure.number) == False
```

**Actual outcome**

```
Matplotlib version: 3.3.0
Traceback (most recent call last):
  File "figure_exists.py", line 11, in <module>
    assert plt.fignum_exists(figure.number) == False
AssertionError

```

**Expected outcome**
No assertion, which is easy to reproduce if one to downgrade to matplotlib-3.2.1:
```
 pip3 install matplotlib==3.2.1 --user; python figure_exists.py 
Collecting matplotlib==3.2.1
  Using cached https://files.pythonhosted.org/packages/b2/c2/71fcf957710f3ba1f09088b35776a799ba7dd95f7c2b195ec800933b276b/matplotlib-3.2.1-cp37-cp37m-manylinux1_x86_64.whl
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/lib/python3.7/site-packages (from matplotlib==3.2.1) (2.4.0)
Requirement already satisfied: numpy>=1.11 in /usr/local/lib64/python3.7/site-packages (from matplotlib==3.2.1) (1.18.4)
Requirement already satisfied: python-dateutil>=2.1 in /usr/lib/python3.7/site-packages (from matplotlib==3.2.1) (2.8.0)
Requirement already satisfied: cycler>=0.10 in /usr/lib/python3.7/site-packages (from matplotlib==3.2.1) (0.10.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/lib64/python3.7/site-packages (from matplotlib==3.2.1) (1.1.0)
Requirement already satisfied: six>=1.5 in /usr/lib/python3.7/site-packages (from python-dateutil>=2.1->matplotlib==3.2.1) (1.12.0)
Requirement already satisfied: setuptools in /usr/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib==3.2.1) (40.8.0)
Installing collected packages: matplotlib
  Found existing installation: matplotlib 3.3.0
    Uninstalling matplotlib-3.3.0:
      Successfully uninstalled matplotlib-3.3.0
Successfully installed matplotlib-3.2.1
Matplotlib version: 3.2.1
```

**Matplotlib version**

  * Operating system: Fedora Core 30
  * Matplotlib version: 3.3.0 (installed via pip as `pip3 install matplotlib==3.3.0 --user` )
  * Matplotlib backend (`print(matplotlib.get_backend())`):
  * Python version: 3.7.0
  * Jupyter version (if applicable):
  * Other libraries:

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This is a partial revert of a small part of #13581.

Fixes #18163.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [n/a] New features are documented, with examples if plot related
- [n/a] Documentation is sphinx and numpydoc compliant
- [n/a] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [n/a] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/_pylab_helpers.py [hunk 1]
```diff
@@ -71,9 +71,10 @@ def destroy(cls, num):
     @classmethod
     def destroy_fig(cls, fig):
         """Destroy figure *fig*."""
-        canvas = getattr(fig, "canvas", None)
-        manager = getattr(canvas, "manager", None)
-        cls.destroy(manager)
+        num = next((manager.num for manager in cls.figs.values()
+                    if manager.canvas.figure == fig), None)
+        if num is not None:
+            cls.destroy(num)
```
Comment (timhoffm) [inline lib/matplotlib/_pylab_helpers.py:77]: Can we add a comment, why we have to destroy by number?

Comment (QuLogic) [reply lib/matplotlib/_pylab_helpers.py:77]: It actually works either way (`manager` or `manager.num`); it's just that the code in `destroy` is simpler for `num`.

**Comment by QuLogic** [comment]
Yea, that's weird, a restart was not effective; I'll try it out on the mac.

**Comment by QuLogic** [comment]
It worked on the Mac Mini, so I restarted Travis again.

**Comment by QuLogic** [comment]
I tried increasing the timeout, which didn't work, but it's appearing on other tests now, so I've reverted that.

**Comment by tacaswell** [comment]
As discussed on the call to get 3.3.1 out the door we are going to disable this test on travis (it should be noted that we do run this test on azure) so the current working theory is that this is a travis-specific issues with the GUI and is not worth the time right now to track down the root problem (but we should open a new issue and label it "good first issue / medium" localized, but involves defeating CI ;) ).

**Pull request merged**